packages/object-utilities/src/js/__protected.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Creates a non-enumerable, non-configurable
+ * and non-writeable property in the given object.
+ * Use this for properties you want to be considered
+ * as protected, functions are public for now.
+ *
+ * Usage:
+ * class WithProtected {
+ *   constructor() {
+ *     // set multiple properties
+ *     __protected(this, { foo: "bar", some: "thing"});
+ *     // set/change one
+ *     __protected(this).foo = "baz";
+ *     // read
+ *     alert(__protected(this).foo);
+ *   }
+ * }
+ *
+ * @param {Object} object – given object
+ * @param {Object} [value] – given value
+ * @return {Object} protected property / context
+ */
+export default function __protected(object, value) {
+  if (object === undefined || object === null) {
+    throw new Error("no object given or null");
+  }
+  if (object.__protected__ === undefined) {
+    Object.defineProperty(object, "__protected__", {
+      configurable: false,
+      enumerable: false,
+      writeable: false,
+      value: {}
+    });
+  }
+  if (value !== undefined) {
```

Comment:
The following block looks very complex, please either provide a comment or what do you think about this structure:

```JS
if(value !== undefined && typeof value === 'object') {
  Object.assign(object.__protected__, value);
}

if (value !== undefined && typeof value !== 'object') {
  throw new Error("value must be an object");
}

return object.__protected__;
```

or

```JS
if(value !== undefined) {
  if (typeof value === 'object') {
    Object.assign(object.__protected__, value);

    return object.__protected__;
  }
  throw new Error("value must be an object");
}

return object.__protected__;
```